### PR TITLE
Rework colibri-imx7 machines

### PR DESCRIPTION
**Pull request by otavio**

### CODE REVIEW

**Comment by otavio** [comment]
@fbertux it looks good to me.

@MaxKrummenacher please take a look at this. If you're Ok with it we can merge it.

File: conf/machine/colibri-imx7-emmc.conf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# The kernel lives in a seperate FAT partition, don't deploy it in /boot/
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base = ""
+
+PREFERRED_PROVIDER_u-boot ?= "u-boot-toradex"
+PREFERRED_PROVIDER_virtual/bootloader ?= "u-boot-toradex"
+PREFERRED_PROVIDER_u-boot-fw-utils ?= "u-boot-toradex-fw-utils"
+UBOOT_MACHINE ?= "colibri_imx7_emmc_defconfig"
+
+IMAGE_FSTYPES += "wic.gz tar.xz"
```
Comment (MaxKrummenacher) [inline conf/machine/colibri-imx7-emmc.conf]: The wic.gz is today already provided by imx-base.inc and could be deleted.

File: conf/machine/colibri-imx7-emmc.conf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# wic support
+WKS_FILE = "sdimage-bootpart.wks"
+WIC_CREATE_EXTRA_ARGS_append = " --no-fstab-update"
+
```
Comment (MaxKrummenacher) [inline conf/machine/colibri-imx7-emmc.conf]: These three lines are sort of provided by imx-base.inc and could be deleted.
As we fuse the SoM to boot from the on module eMMC, first boot area U-Boot is not needed as part of the wic image, however it does also not hurt, so we could switch to the imx-base.inc setting "sdimage-bootpart.wks".

**Comment by MaxKrummenacher** [comment]
Hi Fabio, Hi Otavio

Thanks for adding the machine.
Updating our stuff is on my list for a long time.

The new machine looks good to me. A cosmetic change I report inline. I successfully built one of our images with the moved machine configuration.

I have issues with the renamed colibri-imx7-nand machine though. It rather keep the old name. This requires us and likely our customers to change Documentation, CI, scripts, possibly OE meta data overrides and what not.
If it is important to mark the difference we could add (NAND) to the #@NAME entry similar to the (eMMC) in the eMMC machine?

I'm an advocate for not changing names if not absolutely necessary. (And even then the after effects are usually not worth it ;-) ).

Thanks again.
Max

**Comment by MaxKrummenacher** [comment]
Just found that u-boot-toradex-fw-utils will need a specific fw_env.config. I will prepare a patch.

**Comment by otavio** [comment]
> The new machine looks good to me. A cosmetic change I report inline. I successfully built one of our images with the moved machine configuration.

Great. We will do the changes and update the PR.

> I have issues with the renamed colibri-imx7-nand machine though. It rather keep the old name. This requires us and likely our customers to change Documentation, CI, scripts, possibly OE meta data overrides and what not.
> If it is important to mark the difference we could add (NAND) to the #@name entry similar to the (eMMC) in the eMMC machine?
> 
> I'm an advocate for not changing names if not absolutely necessary. (And even then the after effects are usually not worth it ;-) ).

I understand your point and we will also need to do these adjustments on our side as well, however, we should use clear names as we cannot use a single machine for this specific case.

In the past, we had a similar situation when we merged several SABRE SD machines on a single one and so far, it went well. so far.

Also, we are finishing up the LTS release and it is likely to become a foundation point for the next few years, so the timing couldn't be better.

What do you think?

File: conf/machine/colibri-imx7-emmc.conf (hunk 2 above)
Comment (otavio) [reply conf/machine/colibri-imx7-emmc.conf]: In fact, adding the U-Boot adds an extra dependency and is technically wrong as it is unused. I'd prefer to not write it there due to that.

Comment (MaxKrummenacher) [reply conf/machine/colibri-imx7-emmc.conf]: I am fine with keeping the WKS_FILE here.

**Comment by MaxKrummenacher** [comment]
I really have strong feelings here.

Currently everything for Colibri iMX7 with NAND goes under colibri-imx7, here in OE, in the U-Boot config, the kernel device trees and our minimized kernel defconfigs. For me it is simply not worth the work required to fix up everything related just to have a machine name which reflects somewhat better that this particular machine is for a Colbri iMX7 with NAND.

Of course, if we could start from scratch and new in advance that we will have a NAND and a eMMC variant I would start with a clear machine naming but unfortunately we have a history here.

**Comment by otavio** [comment]
> Currently everything for Colibri iMX7 with NAND goes under colibri-imx7, here in OE, in the U-Boot config, the kernel device trees and our minimized kernel defconfigs.

We won't need to change all internal things, just OE name. Also, we can add the old name as an override so we keep compatibility with existing code.

File: conf/machine/colibri-imx7-emmc.conf (hunk 1 above)
Comment (otavio) [reply conf/machine/colibri-imx7-emmc.conf]: Done.

File: conf/machine/colibri-imx7-nand.conf [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #@TYPE: Machine
-#@NAME: Toradex Colibri iMX7D/S
+#@NAME: Toradex Colibri iMX7D/S (NAND)
 #@SOC: i.MX 7Dual / i.MX 7Solo
-#@DESCRIPTION: Machine configuration for Toradex Colibri iMX7 SOM
+#@DESCRIPTION: Machine configuration for Toradex Colibri iMX7 SOM (NAND)
 #@MAINTAINER: Max Krummenacher <max.krummenacher@toradex.com>
 
 MACHINEOVERRIDES =. "mx7:mx7d:"
```
Comment (otavio) [inline conf/machine/colibri-imx7-nand.conf]: @fbertux we need to add the old machine name as an override so we don't break existing code.

**Pull request merged**